optuna/storages/_base.py

```diff
@@ -209,7 +210,7 @@ def get_study_id_from_name(self, study_name: str) -> int:
         """
         raise NotImplementedError
 
-    @abc.abstractmethod
+    @deprecated("2.9.0", "4.0.0")
```

Comment:
[Question] Could you tell me why you specify `2.9.0` instead of `3.0.0`?

Reply:
Sorry, it should be 3.0.0! Thank you.

Reply:
Thank you for your update. 

By the way, we usually remove deprecated methods after two major releases. For example, `optuna.integration.CmaEsSampler` was deprecated at 2.0.0 and will be removed at 4.0.0.

> optuna/integration/cma.py:@deprecated("2.0.0", "4.0.0", text=_cma_deprecated_msg)

Do you intentionally set the `removed_version` to 4.0.0?

IMO, it looks reasonable to remove this method at 4.0.0 since it is not a user-facing API and no one uses it.

Reply:
@g-votte suggested that this method could be removed without deprecation.
This is mainly because it is no longer used and it is a private API.
What do you think of it @keisuke-umezawa ?

Reply:
I agreed on it. I will remove those methods.